DataProcessing/datax-host/src/main/scala/datax/fs/HadoopClient.scala

```diff
@@ -124,38 +124,46 @@ object HadoopClient {
     * warn if key is not found but we let it continue so static key settings outside of the job can still work
     * @param vaultName key vault name to get the key of storage account
     * @param sa name of the storage account
+    * @param blobStorageKey broadcasted storage account key
     */
-  def resolveStorageAccount(vaultName: String, sa: String) : Option[String] = {
-    // Fetch secret from keyvault using KeyVaultMsiAuthenticatorClient and if that does not return secret then fetch it using secret scope  
-    val secretId = s"keyvault://$vaultName/${ProductConstant.ProductRoot}-sa-$sa"
-    KeyVaultClient.getSecret(secretId) match {
-      case Some(value)=>
-        logger.warn(s"Retrieved key for storage account '$sa' with secretid:'$secretId'")
-        setStorageAccountKey(sa, value)
-        Some(value)
-      case None =>	    
-        val databricksSecretId = s"secretscope://$vaultName/${ProductConstant.ProductRoot}-sa-$sa"
-        KeyVaultClient.getSecret(databricksSecretId) match {
-          case Some(value)=>
-            logger.warn(s"Retrieved key for storage account '$sa' with secretid:'$databricksSecretId'")
-            setStorageAccountKey(sa, value)
-            Some(value)
-          case None =>
-            logger.warn(s"Failed to find key for storage account '$sa' with secretid:'$secretId' and '$databricksSecretId'")
-            None
-        }
+  def resolveStorageAccount(vaultName: String, sa: String, blobStorageKey: broadcast.Broadcast[String] = null) : Option[String] = {
+    if(blobStorageKey != null) {
+      setStorageAccountKey(sa, blobStorageKey.value)
+      Some(blobStorageKey.value)
+    }
+    else {
+      // Fetch secret from keyvault using KeyVaultMsiAuthenticatorClient and if that does not return secret then fetch it using secret scope
+      val secretId = s"keyvault://$vaultName/${ProductConstant.ProductRoot}-sa-$sa"
+      KeyVaultClient.getSecret(secretId) match {
+        case Some(value)=>
+          logger.warn(s"Retrieved key for storage account '$sa' with secretid:'$secretId'")
+          setStorageAccountKey(sa, value)
+          Some(value)
+        case None =>
+          val databricksSecretId = s"secretscope://$vaultName/${ProductConstant.ProductRoot}-sa-$sa"
+          KeyVaultClient.getSecret(databricksSecretId) match {
+            case Some(value)=>
+              logger.warn(s"Retrieved key for storage account '$sa' with secretid:'$databricksSecretId'")
```

Comment:
Same security concern as above, is dataBricksSecretId sensitive data?

Reply:
we create databrickssecretid. not sensitive